### ISSUE TITLE
training code

### ISSUE THREAD

**Issue by jordane95** [MARIO-Math-Reasoning/Super_MARIO#8]
Will the model fine-tuning code also be released soon? Thanks.

**Comment by lovecambi** [comment]
> Will the model fine-tuning code also be released soon? Thanks.

Sorry. Currently, we cannot release the training code. But it's not difficult to implement. Basically, only two modifications are required.

1. Add a value head to the last layer of LLM.
2. auto-regressive loss (positive example) + 0.1 * MSE value loss (positive and negative examples)

**Comment by lovecambi** [comment]
Another hint, we use [trl](https://github.com/huggingface/trl/blob/main/trl/models/modeling_value_head.py) to support value model training.
